numpy/core/src/umath/override.c

```diff
@@ -429,8 +429,19 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
         /* Check if there is a method left to call */
         if (!override_obj) {
             /* No acceptable override found. */
-            PyErr_SetString(PyExc_TypeError,
-                            "__array_ufunc__ not implemented for this type.");
+            static PyObject *errmsg_formatter = NULL;
+            PyObject *errmsg;
+
+            npy_cache_import("numpy.core._internal",
+                             "array_ufunc_errmsg_formatter",
+                             &errmsg_formatter);
+            errmsg = PyObject_Call(errmsg_formatter, override_args,
```

Comment:
Note that `npy_cache_import` may fail, in which case `errmsg_formatter` will be NULL.